ui/admin/app/components/agent/shared/AgentEnvSection.tsx

```diff
@@ -0,0 +1,75 @@
+import { toast } from "sonner";
+
+import { Agent } from "~/lib/model/agents";
+import { Workflow } from "~/lib/model/workflows";
+import { EnvironmentApiService } from "~/lib/service/api/EnvironmentApiService";
+
+import { EnvForm } from "~/components/agent/shared/AgentEnvironmentVariableForm";
+import { Button } from "~/components/ui/button";
+import {
+    Dialog,
+    DialogContent,
+    DialogDescription,
+    DialogHeader,
+    DialogTitle,
+    DialogTrigger,
+} from "~/components/ui/dialog";
+import { useAsync } from "~/hooks/useAsync";
+
+type AgentEnvFormProps = {
+    entity: Agent | Workflow;
+    entityType: "agent" | "workflow";
+};
+
+export function AgentEnvSection({ entity, entityType }: AgentEnvFormProps) {
```

Comment:
nit: super minor nit. Looks like workflow or agent can be passed as the entity, wondering if this is a generic `EnvSection` instead of `AgentEnvSection` Maybe same thing for AgentEnvironmentVariableForm?

Reply:
I've been going back and forth on the same thing... Will update